[
  {
    "project": "seandstewart/typical",
    "commit": "adce0ae6d59b1c6077f526daa765e11ebd637bee",
    "filename": "typic/constraints/error.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/seandstewart-typical/typic/constraints/error.py",
    "file_hunks_size": 1,
    "min_patch_found": true,
    "single_hunk": true,
    "fit_TFix": false,
    "delete_only_patch": true,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "typic/constraints/error.py:17:33 Incompatible variable type [9]: _from is declared to have type `Exception` but is used as type `None`.",
    "message": " _from is declared to have type `Exception` but is used as type `None`.",
    "rule_id": "Incompatible variable type [9]",
    "warning_line_no": 17,
    "warning_line": "def raise_exc(exc: Exception, *, _from: Exception = None):",
    "min_patch": [
      {
        "hunk_fit_TFix": false,
        "inside_window": true,
        "delete_only": true,
        "has_suppression": false,
        "source_code": "    pass\n\n\ndef raise_exc(exc: Exception, *, _from: Exception = None):\n    raise exc from _from\n",
        "source_code_len": 95,
        "target_code": "    pass\n",
        "target_code_len": 9,
        "diff_format": "@@ -14,5 +14,1 @@\n     pass\n-\n-\n-def raise_exc(exc: Exception, *, _from: Exception = None):\n-    raise exc from _from\n",
        "source_code_with_indent": "    pass\n\n\n<DED>def raise_exc(exc: Exception, *, _from: Exception = None):\n    <IND>raise exc from _from\n",
        "source_code_with_indent_exact_match": true,
        "target_code_with_indent": "    pass\n",
        "target_code_with_indent_exact_match": true
      }
    ]
  }
]